cpp/src/io/csv/writer_impl.cu

```diff
@@ -160,8 +166,9 @@ struct column_to_strings_fn {
   std::enable_if_t<std::is_same_v<column_type, bool>, std::unique_ptr<column>> operator()(
     column_view const& column) const
   {
-    return cudf::strings::detail::from_booleans(
-      column, options_.get_true_value(), options_.get_false_value(), stream_, mr_);
+    string_scalar true_string{options_.get_true_value(), true, stream_, mr_};
+    string_scalar false_string{options_.get_false_value(), true, stream_, mr_};
```

Comment:
The user-provided memory resource should only be used for the buffers that are returned to the user. Internally we should always use the default mr.
```suggestion
    string_scalar true_string{options_.get_true_value(), true, stream_};
    string_scalar false_string{options_.get_false_value(), true, stream_};
```

Reply:
Thanks, good to know!